### PR TITLE
Simplify component management

### PR DESCRIPTION
**Pull request by tehasdf**
The over-arching idea: put all component creation logic in one place.

Instead of having the service order in one place, their components in another,
actually instantiating the classes somewhere else, and on top of that deciding
whether we should skip installation or not in yet another place, have all
of that in a single function.

Admittedly, the skip logic makes that function be not the most beautiful thing
in the world, but now it at least fits on half on a screen, instead of being
split between fifty files.

### CODE REVIEW

File: cfy_manager/main.py [hunk 1]
```diff
@@ -632,6 +631,62 @@ def _create_component_objects(include_components):
         )
 
 
+def _get_components(include_components=None):
+    """Get the component objects based on the config.
+
+    This looks at the config, and returns only the component objects
+    that are supposed to be installed(/configured/started).
+    """
+    c = []
```
Comment (tehasdf) [inline cfy_manager/main.py:617]: if you can figure out a better name for this, let me know

File: cfy_manager/components/globals.py [hunk 2]
```diff
@@ -101,6 +104,22 @@ def _set_hostname():
...
+            and DATABASE_SERVICE in config[SERVICES_TO_INSTALL]
+        )
+        or (
+            DATABASE_SERVICE not in config[SERVICES_TO_INSTALL]
+            and MANAGER_SERVICE in config[SERVICES_TO_INSTALL]
+        )
+    ):
+        config[POSTGRESQL_SERVER][SSL_ENABLED] = True
+        config[POSTGRESQL_SERVER][ENABLE_REMOTE_CONNECTIONS] = True
+        config[POSTGRESQL_CLIENT][SSL_ENABLED] = True
```
Comment (geokala) [inline cfy_manager/components/globals.py:120]: We could eliminate two of these by making TLS be always-on on the DB.

Comment (geokala) [reply cfy_manager/components/globals.py:120]: And maybe we could check with Jonathan/Isaac whether ENABLE_REMOTE_CONNECTIONS needs to be kept- AFAIK the only use for that is if you want an AIO manager to allow external DB conns?

File: cfy_manager/main.py (hunk 1 above)
Comment (geokala) [reply cfy_manager/main.py:617]: 'required' or 'required_components'?

File: cfy_manager/main.py [hunk 3]
```diff
@@ -602,34 +606,80 @@ def _validate_components_prepared(cmd):
...
+    return c
+
+
+def _filter_components(components, include_components):
+    """Filter the components list based on the includes given by the user.
+
+    This allows `cfy_manager start --include-components amqp_postgres`,
+    which should only then start amqppostgres and nothing else.
+
+    This trnaslates "amqp_postgres" -> "amqppostgres", and then filters
```
Comment (geokala) [inline cfy_manager/main.py:673]: translates

File: cfy_manager/main.py [hunk 4]
```diff
@@ -602,34 +606,80 @@ def _validate_components_prepared(cmd):
...
-    components_to_install = _get_components_list(include_components)
-    for component_name in components_to_install:
-        component_config = config.get(component_name, {})
-        skip_installation = component_config.get('skip_installation', False)
-        components.append(
-            ComponentsFactory.create_component(component_name,
-                                               skip_installation)
-        )
+    include_components = {
+        ''.join(name.lower().split('_')) for name in include_components
```
Comment (geokala) [inline cfy_manager/main.py:677]: .replace('_', '')?

File: cfy_manager/main.py (hunk 1 above)
Comment (tehasdf) [reply cfy_manager/main.py:617]: but they're not really required, are they :/
they're even filtered out later

File: cfy_manager/components/globals.py (hunk 2 above)
Comment (tehasdf) [reply cfy_manager/components/globals.py:120]: i'm only moving this function, so i'm going to declare any actual TLS changes outside the scope of this code-level refactor

File: cfy_manager/main.py (hunk 4 above)
Comment (tehasdf) [reply cfy_manager/main.py:677]: oh yeah duh

File: cfy_manager/main.py (hunk 1 above)
Comment (geokala) [reply cfy_manager/main.py:617]: current_components?

Comment (geokala) [reply cfy_manager/main.py:617]: I mean, what are they? Configured ones?

File: cfy_manager/components/globals.py (hunk 2 above)
Comment (geokala) [reply cfy_manager/components/globals.py:120]: Fair enough, will mark this thread as resolved.

**Pull request merged**